packages/core/src/install/createPeerDependencyPatcher.ts

```diff
@@ -21,7 +21,19 @@ export default function (
         if (peerDependencyRules.allowedVersions[peerName] === '*') {
           pkg.peerDependencies![peerName] = '*'
         } else {
-          pkg.peerDependencies![peerName] += ` || ${peerDependencyRules.allowedVersions[peerName]}`
+          const allowedVersions = peerDependencyRules.allowedVersions[peerName].split(' || ')
+
+          allowedVersions.forEach(allowedVersion => {
+            const allowedVersionPrefix = ` || ${allowedVersion}`
```

Comment:
this will also match `1 || 2.3` for allowed version `2`

Reply:
a fine point. I'll fix it and just split them all.

Reply:
Done - simplified the code, added checks for version ranges that have no space around the `||`, and a test for the partial match.